lua/xbase/init.lua

```diff
@@ -28,6 +28,46 @@ local setup_lsp = function(opts)
   }))
 end
 
+local function try_attach_code_actions(opts)
+    if not config.values.code_actions.enable then
+        return
+    end
+    local swift_actions = require("xbase.treesitter")
+    if swift_actions == nil then
+        return
+    end
+    local attach_xbase_code_actions = require("xbase.null_ls")
+    if attach_xbase_code_actions == nil then
+        return
+    end
+    local code_actions = {}
+    if opts.code_actions.use_builtin_actions then
```

Comment:
Hmmm somehow I got errored here init.lua:44: attempt to index local 'opts' (a nil value).

Oh I totally forgot that try_attach in the autocmd.

try_attach_code_actions needs to be called outside try_attach. It needs be called in setup function once

My bad

Reply:
after config.set called in setup:
`try_attach_code_actions(config.values)`